pkg/sql/tablewriter.go

```diff
@@ -226,7 +226,10 @@ func (tu *tableUpserter) init(txn *client.Txn) error {
 	tu.indexKeyPrefix = sqlbase.MakeIndexKeyPrefix(tu.tableDesc, tu.tableDesc.PrimaryIndex.ID)
 
 	allColsIdentityExpr := len(tu.ri.insertCols) == len(tu.tableDesc.Columns) &&
-		tu.evaler != nil && tu.evaler.isIdentityEvaler()
+		tu.evaler != nil && tu.evaler.isIdentityEvaler() &&
+		// Disable the upsert fast path when columns or indexes are being added
+		// or dropped.
```

Comment:
This is mildly unfortunate as the fast-path is ~2x faster than the slow-path. So adding a column will cause performance to temporarily drop in half? Is there any hope to fix this eventually? Perhaps file an issue.